### PR TITLE
[master-2.x] Fix Transaction verify

### LINKED ISSUE #1999: [master-2.x] A bug at References when Neo2 transfer UTXO
**Describe the bug**
In
https://github.com/neo-project/neo/blob/e18d431760aaa103de5200cbfe3f45aa693361e6/neo/Network/P2P/Payloads/Transaction.cs#L92-L104

`Output = tx.Outputs[p.PrevIndex]`

If tx.Outputs length is 1 and p.PrevIndex is 1, there will be `Output = tx.Outputs[1]` rather than `Output = tx.Outputs[0]`,so there 
 will thrown an exception, this exception will cause RPC's `sendrawtransaction` method to stop working.

**To Reproduce**
1. Start a Neo2 node;
2. Structuring a UTXO transaction like:
```C#
var inputs = new List<CoinReference>
{
    new CoinReference()
    {
        PrevHash = prevHash,
        PrevIndex = 1
    }
}.ToArray();

var outputs = new List<TransactionOutput>
{
    new TransactionOutput()
    {
        AssetId = Blockchain.GoverningToken.Hash,
        ScriptHash = hash,
        Value = value
    }
}.ToArray();
```
...

3. Use ‘sendrawtransaction’ of RPC to send this transaction's hex string.

Hex string:
```
d1003314ac61af17b4970b71e7a0317fb50d04b85dde410751c106726566756e64670ef656d72483fab3804c41ea0f052dab8138da170220ac61af17b4970b71e7a0317fb50d04b85dde4107f1207358baa8ddab0186203f06cdf67f9c8e9f90554269fce6d5b5aaa43e792e78840107512de83ec6dd6018f6567591a5e81ecaf68c88b9f9af92394728c19720a9e00100019b7cffdaa674beae0f930ebe6085af9093e5fe56b34a5c220ccdcf6efc336fc500e1f505000000000ef656d72483fab3804c41ea0f052dab8138da17024140604cb63f5f4306124599327ecb6232fe420f22c0a36bc9ebdd2b751d3dc152ce0ef1addcb26b4deee4c1741ee868cb76d127cff898ec971d7ffe8689830905a7232102d9552f201a06e4103a90bdae0caec2def340d721a298bbed3b635d6f70f3dc01ac0352013100
```

**Expected behavior**
Handle the exception and let the `sendrawtransaction` can continue working.

**Screenshots**
Printed exception:
image

**Platform:**
 - OS: All
 - Version [neo master 2.x]

### PR DESCRIPTION
**Pull request by Qiao-Jin**
Close #1999

### CODE REVIEW

**Comment by ProDog** [comment]
Tesed OK.

request:
```
{
  "jsonrpc": "2.0",
  "id": 1,
  "method": "sendrawtransaction",
  "params": ["d1003314ac61af17b4970b71e7a0317fb50d04b85dde410751c106726566756e64670ef656d72483fab3804c41ea0f052dab8138da170220ac61af17b4970b71e7a0317fb50d04b85dde4107f1207358baa8ddab0186203f06cdf67f9c8e9f90554269fce6d5b5aaa43e792e78840107512de83ec6dd6018f6567591a5e81ecaf68c88b9f9af92394728c19720a9e00100019b7cffdaa674beae0f930ebe6085af9093e5fe56b34a5c220ccdcf6efc336fc500e1f505000000000ef656d72483fab3804c41ea0f052dab8138da17024140604cb63f5f4306124599327ecb6232fe420f22c0a36bc9ebdd2b751d3dc152ce0ef1addcb26b4deee4c1741ee868cb76d127cff898ec971d7ffe8689830905a7232102d9552f201a06e4103a90bdae0caec2def340d721a298bbed3b635d6f70f3dc01ac0352013100"]
}
```

result:
```
{
    "jsonrpc": "2.0",
    "id": 1,
    "error": {
        "code": -504,
        "message": "Block or transaction validation failed."
    }
}
```

**Review by shargon — COMMENTED** [review]
What about add the try/catch in https://github.com/neo-project/neo/blob/a3217ea6d3cb75d5c5b07600cedd3f9a0b3d103b/neo/Network/P2P/Payloads/Transaction.cs#L99 and return null

**Comment by KickSeason** [comment]
`base.Verify()` before check `NetworkFee`?

**Comment by Qiao-Jin** [comment]
> What about add the try/catch in
> 
> https://github.com/neo-project/neo/blob/a3217ea6d3cb75d5c5b07600cedd3f9a0b3d103b/neo/Network/P2P/Payloads/Transaction.cs#L99
> 
> and return null

Fixed

File: neo/Network/P2P/Payloads/Transaction.cs [hunk 1]
```diff
@@ -269,61 +271,54 @@ bool IInventory.Verify(Snapshot snapshot)
...
-                        if (results_issue.Length > 0)
-                            return false;
-                        break;
-                }
-                if (Attributes.Count(p => p.Usage == TransactionAttributeUsage.ECDH02 || p.Usage == TransactionAttributeUsage.ECDH03) > 1)
-                    return false;
-                if (!VerifyReceivingScripts()) return false;
-                return this.VerifyWitnesses(snapshot);
             }
-            catch (Exception)
```
Comment (shargon) [inline neo/Network/P2P/Payloads/Transaction.cs]: This catch was generic, so it could cover more cases than indexOfOutBounds

**Comment by ProDog** [comment]
Tesed OK.

**Comment by Qiao-Jin** [comment]
@shargon @erikzhang Can we merge this one?

**Review by shargon — APPROVED** [review]

**Pull request merged**